### PR TITLE
Updating the CONTRIBUTING.md to match what we have in Helm, I also ad…

### PR DESCRIPTION
**Pull request by robertsirc**
Updating the CONTRIBUTING.md to match what we have in Helm and added a section to denote the requirements for GPG.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +1,336 @@
...
+### PR Specific
+
+| Label | Description |
+| ----- | ----------- |
+| `awaiting review` | Indicates a PR has been triaged and is ready for someone to review |
+| `breaking` | Indicates a PR has breaking changes (such as API changes) |
+| `in progress` | Indicates that a maintainer is looking at the PR, even if no review has been posted yet |
+| `needs rebase` | Indicates a PR needs to be rebased before it can be merged |
+| `needs pick` | Indicates a PR needs to be cherry-picked into a feature branch (generally bugfix branches). Once it has been, the `picked` label should be applied and this one removed |
+| `picked` | This PR has been cherry-picked into a feature branch |
```
Comment (mattfarina) [inline CONTRIBUTING.md:313]: The labels are different over here. Not as complex. You can find them at https://github.com/helm/helm-mapkubeapis/labels. They can all be considered common. Can you update for this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,3 +1,336 @@
...
+far as writing a proposal is meant to save the potential author time. Many ideas have been proposed;
+it's quite likely there are others in the community who may be working on a similar proposal, or a
+similar proposal may have already been written.
+
+HIPs are submitted to the [helm/community repository](https://github.com/helm/community). [HIP
+1](https://github.com/helm/community/blob/master/hips/hip-0001.md) describes the process to write a
+HIP as well as the review process.
+
+After your proposal has been approved, follow the [developer's
+guide](https://helm.sh/docs/community/developers/) to get started.
```
Comment (mattfarina) [inline CONTRIBUTING.md:207]: You can drop this section. This plugin isn't going to go outside of scope of it's existing place so something a HIP would deal with won't happen here.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,3 +1,336 @@
...
+3. Submit a pull request, making sure to sign your work and link the related issue.
+
+Coding conventions and standards are explained in the [official developer
+docs](https://helm.sh/docs/developers/).
+
+## Pull Requests
+
+Like any good open source project, we use Pull Requests (PRs) to track code changes.
+
+### PR Lifecycle
```
Comment (mattfarina) [inline CONTRIBUTING.md:223]: Likely drop this section and PRs are much simpler here.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,3 +1,336 @@
...
+
+Documentation PRs will follow the same lifecycle as other PRs. They will also be labeled with the
+`docs` label. For documentation, special attention will be paid to spelling, grammar, and clarity
+(whereas those things don't matter *as* much for comments in code).
+
+## The Triager
+
+Each week, one of the core maintainers will serve as the designated "triager" starting after the
+public stand-up meetings on Thursday. This person will be in charge triaging new PRs and issues
+throughout the work week.
```
Comment (mattfarina) [inline CONTRIBUTING.md:277]: We don't have a triager. We can drop this section.

**Review by mattfarina — CHANGES_REQUESTED** [review]
Thanks for the PR. This plugin is MUCH simpler in terms of workflow than Helm itself. The sign-off, signing, and size labels are good. Most of the rest of the workflow stuff can be removed as it doesn't happen on this plugin.

**Comment by robertsirc** [comment]
@mattfarina all your changes have been made.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,3 +1,227 @@
...
+      that they are taking it.
+    - `proposal` and `support/question` issues should stay open until resolved or if they have not
+      been active for more than 30 days. This will help keep the issue queue to a manageable size
+      and reduce noise. Should the issue need to stay open, the `keep open` label can be added.
+4. Issue closure
+
+## How to Contribute a Patch
+
+1. Identify or create the related issue. If you're proposing a larger change to
+   Helm, see [Proposing an Idea](#proposing-an-idea).
```
Comment (mattfarina) [inline CONTRIBUTING.md:176]: There is no section on "Proposing an Idea". I think you can just remove "1." and only have a 2 item list.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,3 +1,227 @@
...
+
+## Issues
+
+Issues are used as the primary method for tracking anything to do with the Helm project.
+
+### Issue Types
+
+There are 5 types of issues (each with their own corresponding [label](#labels)):
+
+- `question/support`: These are support or functionality inquiries that we want to have a record of
```
Comment (mattfarina) [inline CONTRIBUTING.md:130]: This one is just "question".

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,3 +1,227 @@
...
+  for future reference. Generally these are questions that are too complex or large to store in the
+  Slack channel or have particular interest to the community as a whole. Depending on the
+  discussion, these can turn into `feature` or `bug` issues.
+- `proposal`: Used for items (like this one) that propose a new ideas or functionality that require
+  a larger community discussion. This allows for feedback from others in the community before a
+  feature is actually  developed. This is not needed for small additions. Final word on whether or
+  not a feature needs a proposal is up to the core maintainers. All issues that are proposals should
+  both have a label and an issue title of "Proposal: [the rest of the title]." A proposal can become
+  a `feature` and does not require a milestone.
```
Comment (mattfarina) [inline CONTRIBUTING.md:139]: There is no "proposal" type.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,3 +1,227 @@
...
+- `feature`: These track specific feature requests and ideas until they are complete. They can
+  evolve from a `proposal` or can be submitted individually depending on the size.
```
Comment (mattfarina) [inline CONTRIBUTING.md:141]: This one is "enhancement"

File: CONTRIBUTING.md (hunk 6 above)
Comment (robertsirc) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 7 above)
Comment (robertsirc) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 5 above)
Comment (robertsirc) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 8 above)
Comment (robertsirc) [reply CONTRIBUTING.md]: done

**Review by scottrigby — COMMENTED** [review]
@robertsirc thanks for this! 

It somehow feels wrong to duplicate so much from https://github.com/helm/helm/blob/main/CONTRIBUTING.md 🤔 But given that sections of helm core don't apply here, I can see why this was done. 

Hear me out… what if this doc starts off by saying something like

>While this [helm-mapkubeapis](https://github.com/helm/helm-mapkubeapis) plugin draws its contributing workflow steps from [Helm core contributing](https://github.com/helm/helm/blob/main/CONTRIBUTING.md), this plugin's workflow is much simpler. So we will link only to the sections from Helm core contributing doc that apply to this plugin. There are several exceptions, where we need to make changes from the corresponding steps in Helm core:
>- "Reporting a Security Issue" - link differs (the correct issue queue for this plugin)
>- "How to Contribute a Patch" - this plugin has a limited scope, and does not need to use the HIP process
>- "Pull Requests" - Helm core has a necessarily complex PR lifecycle, while this Plugin uses a standard maintainer PR review process
>- "Labels" - this plugin has different labels for a simpler workflow

Then - keeping your headers below - replace all other section content with a link to the applicable sections in Helm core, like:

```md
## Sign-off On Your Work

See https://github.com/helm/helm/blob/main/CONTRIBUTING.md#sign-your-work

## Support Channels

See https://github.com/helm/helm/blob/main/CONTRIBUTING.md#support-channels

## Semantic Versioning

See https://github.com/helm/helm/blob/main/CONTRIBUTING.md#semantic-versioning

## Issues

See https://github.com/helm/helm/blob/main/CONTRIBUTING.md#issues

etc…
```

WDYT?

PS, either way, I think we should add the "Signing Your Work" (GPG signing) section – and clarifying header change "Sign Your Work" ->  "Sign-off On Your Work" – upstream to Helm core CONTRIBUTING.md.

**Comment by robertsirc** [comment]
I am ok with the approach. @mattfarina thoughts?

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,3 +1,218 @@
...
+For a quick summary of our backward compatibility guidelines for releases between 3.0 and 4.0:
+
+- Command line commands, flags, and arguments MUST be backward compatible
+- File formats (such as Chart.yaml) MUST be backward compatible
+- Any chart that worked on a previous version of Helm 3 MUST work on a new version of Helm 3
+  (barring the cases where (a) Kubernetes itself changed, and (b) the chart worked because it
+  exploited a bug)
+- Chart repository functionality MUST be backward compatible
+- Go libraries inside of `pkg/` MUST remain backward compatible, though code inside of `cmd/` and
+  `internal/` may be changed from release to release without notice.
```
Comment (mattfarina) [inline CONTRIBUTING.md:120]: This is really for Helm itself (and refers to 3.0 and 4.0 while mapkubeapis isn't even 1.0). Can you delete it?

Comment (robertsirc) [reply CONTRIBUTING.md]: done and updated.

**Review by mattfarina — APPROVED** [review]
@scottrigby I think this is ok to merge as is. It's different enough from helm/helm that pointing to helm/helm will just confuse people. It has not followed the same conventions as helm/helm.

**Pull request merged**